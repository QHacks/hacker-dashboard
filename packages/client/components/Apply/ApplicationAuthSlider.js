import React, { PureComponent } from "react";
import { css } from "emotion";

class ApplicationAuthSlider extends PureComponent {
  changeSelected(index) {
    this.props.changeSelected(index);
  }

  render() {
    const sliderWidth = Math.trunc(100000 / this.props.items.length) / 1000;

    const sliderCss = css({
      position: "absolute",
      left: `${this.props.selectedItem * sliderWidth}%`,
      top: 0,
      height: "44px",
      backgroundColor: "#C81C2E",
      borderRadius: "44px",
      width: `${sliderWidth}%`,
      zIndex: 2,
      transition: "left 0.8s ease"
    });

    const navBarCss = css({
      display: "grid",
      gridTemplateColumns: `repeat(${this.props.items.length}, 1fr)`,
      width: "66%",
      "@media screen and (max-width: 760px)": { width: "100%" },
      maxWidth: "600px",
      margin: "20px auto 35px auto",
      backgroundColor: "#F6F6F6",
      height: "44px",
      position: "relative",
      zIndex: 2,
      border: "1px solid #e4e4e4",
      borderRadius: "44px",
      "> button.item": {
        textAlign: "center",
        lineHeight: "44px",
        textTransform: "uppercase",
        cursor: "pointer",
        background: "none",
        border: "none",
        fontSize: "18px",
        zIndex: 3,
        transition: "color 1s ease"
      },
      "> button.item.selectedItem": {
        color: "white"
      },
      "> button.item:not(.selectedItem)": {
        color: "#4d4d4d"
      },
      "> button.item:not(.selectedItem):hover": {
        color: "#C81C2E"
      }
    });

    const items = this.props.items.map((item, i) => (
      <button
        key={item}
        onClick={() => this.changeSelected(i)}
        className={this.props.selectedItem === i ? "selectedItem item" : "item"}
      >
        {item}
      </button>
    ));

    return (
      <div className={navBarCss}>
        {items}
        <div role="none presentation" className={sliderCss} />
      </div>
    );
  }
}

export default ApplicationAuthSlider;
