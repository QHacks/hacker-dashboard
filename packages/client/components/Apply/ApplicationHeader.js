import React, { Component } from "react";
import { offWhite } from "../../assets/colors";

class ApplicationHeader extends Component {
  render() {
    return (
      <div
        css={`
          background-color: ${offWhite};
          height: 325px;
          text-align: center;
          padding-top: 155px;
          background: url(${require("../../assets/img/circuits.png")}) no-repeat
            center center;
          background-size: cover;
          text-transform: uppercase;
        `}
      >
        <h1>QHacks 2019</h1>
        <h2>Hacker Application</h2>
      </div>
    );
  }
}

export default ApplicationHeader;
