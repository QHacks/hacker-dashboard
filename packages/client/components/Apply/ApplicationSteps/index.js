import MediaQuery from "react-responsive";
import React from "react";
import ApplicationStepsMobile from "./ApplicationStepsMobile";
import ApplicationStepsDesktop from "./ApplicationStepsDesktop";
import { mobileMaxWidth, desktopMinWidth } from "../../../assets/constants";

export default (props) => (
  <>
    <MediaQuery maxWidth={mobileMaxWidth}>
      <ApplicationStepsMobile {...props} />
    </MediaQuery>
    <MediaQuery minWidth={desktopMinWidth}>
      <ApplicationStepsDesktop {...props} />
    </MediaQuery>
  </>
);
