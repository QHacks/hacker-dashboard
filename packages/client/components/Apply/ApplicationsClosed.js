import React from "react";

export default () => (
  <div>
    <p>
      We've closed applications for 2018, but we hope to see you next winter!{" "}
    </p>
    <p>
      <a href="https://qhacks.io">
        Go to our website for information about next year!
      </a>
    </p>
  </div>
);
