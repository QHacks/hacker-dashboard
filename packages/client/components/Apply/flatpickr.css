.flatpickr-calendar {
  font-family: "Encode Sans", sans-serif !important;
}
