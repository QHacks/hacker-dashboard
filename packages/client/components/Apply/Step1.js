import React from "react";
import { Link } from "react-router-dom";
import FinePrint from "../FinePrint/FinePrint";
import ApplicationAuthSlider from "./ApplicationAuthSlider";
import ValidationError from "../ValidationError/ValidationError";
import ActionButton from "../ActionButton/ActionButton";
import { grey } from "../../assets/colors";

const Step1 = (props) => (
  <div
    css={`
      ${props.formStyle}
      text-align: center;
    `}
  >
    <h3>
      {props.returningHacker ? "Welcome back!" : "Join the QHacks family!"}
    </h3>
    <ApplicationAuthSlider
      items={["new hacker", "returning hacker"]}
      changeSelected={(i) => props.changeSelected(i)}
      selectedItem={props.returningHacker ? 1 : 0}
    />
    {props.returningHacker ? (
      <div>
        <h2 css={props.authHeadingStyle}>Login to Account</h2>
        <div css={props.subsectionStyle}>
          <div
            css="
                width: 100%;
              "
          >
            <section css={props.inputRowStyle}>
              <div>
                <label htmlFor="email">Email</label>
                <input
                  type="text"
                  placeholder="Email address"
                  value={props.authAnswers.email}
                  onChange={(e) => props.setAuthAnswer("email", e.target.value)}
                />
                <ValidationError message={props.errors.email} />
              </div>
            </section>
            <section css={props.inputRowStyle}>
              <div>
                <label htmlFor="password">Password</label>
                <input
                  type="password"
                  placeholder="Password"
                  value={props.authAnswers.password}
                  onChange={(e) =>
                    props.setAuthAnswer("password", e.target.value)
                  }
                />
                <ValidationError message={props.errors.password} />
              </div>
            </section>
            <section css={props.inputRowStyle}>
              <div>
                <Link
                  css={`
                    font-weight: 600;
                    color: ${grey};
                    text-decoration: underline;
                    float: right;
                  `}
                  to="/forgot-password"
                >
                  Forgot password
                </Link>
                <br />
                <ActionButton
                  disabled={props.hasErrors}
                  style="margin-top: 50px;"
                  color="red"
                  onClick={() => props.nextStep()}
                >
                  Login
                </ActionButton>
              </div>
            </section>
          </div>
        </div>
      </div>
    ) : (
      <div>
        <h2 css={props.authHeadingStyle}>Create an Account</h2>
        <div css={props.subsectionStyle}>
          <div
            css="
                width: 100%;
              "
          >
            <section css={props.inputRowStyle}>
              <div>
                <label htmlFor="firstName">First Name</label>
                <input
                  type="text"
                  id="firstName"
                  placeholder="First name"
                  key="firstName"
                  value={props.authAnswers.firstName}
                  onChange={(e) =>
                    props.setAuthAnswer("firstName", e.target.value)
                  }
                />
                <ValidationError message={props.errors.firstName} />
              </div>
              <div>
                <label htmlFor="lastName">Last Name</label>
                <input
                  type="text"
                  id="lastName"
                  key="lastName"
                  placeholder="Last name"
                  value={props.authAnswers.lastName}
                  onChange={(e) =>
                    props.setAuthAnswer("lastName", e.target.value)
                  }
                />
                <ValidationError message={props.errors.lastName} />
              </div>
            </section>
            <section css={props.inputRowStyle}>
              <div>
                <label htmlFor="email">Email</label>
                <input
                  id="email"
                  key="email"
                  type="text"
                  placeholder="Email address"
                  value={props.authAnswers.email}
                  onChange={(e) => props.setAuthAnswer("email", e.target.value)}
                />
                <ValidationError message={props.errors.email} />
              </div>
            </section>
            <section css={props.inputRowStyle}>
              <div>
                <label htmlFor="password">Password</label>
                <input
                  id="password"
                  key="password"
                  type="password"
                  placeholder="Password"
                  value={props.authAnswers.password}
                  onChange={(e) => {
                    props.resetConfirmPassword();
                    props.setAuthAnswer("password", e.target.value);
                  }}
                />
                <ValidationError message={props.errors.password} />
              </div>
            </section>
            <section css={props.inputRowStyle}>
              <div>
                <label htmlFor="confirmPassword">Confirm password</label>
                <input
                  id="confirmPassword"
                  key="confirmPassword"
                  placeholder="Confirm password"
                  type="password"
                  value={props.authAnswers.confirmPassword}
                  onChange={(e) =>
                    props.setAuthAnswer("confirmPassword", e.target.value)
                  }
                />
                <ValidationError message={props.errors.confirmPassword} />
              </div>
            </section>
            <section css={props.inputRowStyle}>
              <div>
                <ActionButton
                  disabled={props.hasErrors}
                  style="margin-top: 50px;"
                  color="red"
                  onClick={() => props.nextStep()}
                >
                  Create Account
                </ActionButton>
              </div>
            </section>
            <FinePrint />
          </div>
        </div>
      </div>
    )}
  </div>
);

export default Step1;
