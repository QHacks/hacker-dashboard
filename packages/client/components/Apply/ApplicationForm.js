import escapeStringRegexp from "escape-string-regexp";
import { graphql, compose } from "react-apollo";
import React, { Component } from "react";
import gql from "graphql-tag";
import axios from "axios";

import ContentWrapper from "../ContentWrapper/ContentWrapper";
import StatusReport from "../StatusReport/StatusReport";
import { steel } from "../../assets/colors";
import { SERVER_HOST } from "../../Client";
import Step1 from "./Step1";
import Step2 from "./Step2";
import Step3 from "./Step3";
import Step4 from "./Step4";

const LOGIN_MUTATION = gql`
  mutation login($input: LoginInput!) {
    login(input: $input) @client
  }
`;

class ApplicationForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      returningHacker: false,
      authAnswers: {
        password: "",
        confirmPassword: "",
        email: "",
        firstName: "",
        lastName: ""
      },
      applicationAnswers: {
        // if you modify this, remember to update the validators
        phoneNumber: "",
        birthday: "",
        program: "",
        graduationYear: "",
        numAttendedHackathons: "",
        fromLocation: "",
        gender: "",
        race: "",
        school: "",
        degree: "",
        graduationMonth: "",
        whyApply: "",
        dreamProject: "",
        passionProject: "",
        agreeMlhCodeConduct: "",
        agreeMlhTosAndPrivacyPolicy: "",
        github: "",
        linkedIn: "",
        personalWebsite: ""
      },
      errors: {},
      hasErrors: false,
      alert: {
        type: "",
        status: null,
        message: ""
      },
      alertShown: false
    };

    this.setApplicationAnswer = this.setApplicationAnswer.bind(this);
    this.changeSelected = this.changeSelected.bind(this);
    this.setAuthAnswer = this.setAuthAnswer.bind(this);
    this.nextStep = this.nextStep.bind(this);
    this.previousStep = this.previousStep.bind(this);
    this.resetConfirmPassword = this.resetConfirmPassword.bind(this);
    this.submit = this.submit.bind(this);
    this.login = this.login.bind(this);
    this.createAccount = this.createAccount.bind(this);
  }

  async login() {
    if (this.validateAllAnswers()) {
      const { email, password } = this.state.authAnswers;

      try {
        const response = await axios.post(`${SERVER_HOST}/oauth/session`, {
          email,
          password,
          grantType: "password"
        });

        this.props.login({
          variables: {
            input: {
              accessToken: response.data.accessToken,
              refreshToken: response.data.refreshToken
            }
          }
        });

        this.nextStep();
      } catch (err) {
        this.setState({
          alert: {
            type: "danger",
            message: "Login failed",
            status: null
          },
          alertShown: true
        });
        setTimeout(() => this.setState({ alertShown: false }), 5000);
      }
    }
  }

  async createAccount() {
    if (this.validateAllAnswers()) {
      const { firstName, lastName, email, password } = this.state.authAnswers;

      try {
        const response = await axios.post(`${SERVER_HOST}/oauth/signup`, {
          firstName,
          lastName,
          email,
          password
        });

        this.props.login({
          variables: {
            input: {
              accessToken: response.data.accessToken,
              refreshToken: response.data.refreshToken
            }
          }
        });

        this.nextStep();
      } catch (err) {
        this.setState({
          alert: {
            type: "danger",
            message: "Create account failed",
            status: null
          },
          alertShown: true
        });
        setTimeout(() => this.setState({ alertShown: false }), 5000);
      }
    }
  }

  validators() {
    return {
      linkedIn: {
        regex: /^.*$/,
        stepNum: 1,
        message: "Please enter a valid URL"
      },
      github: {
        regex: /^.*$/,
        stepNum: 1,
        message: "Please enter a valid URL"
      },
      personalWebsite: {
        regex: /^.*$/,
        stepNum: 1,
        message: "Please enter a valid URL"
      },
      email: {
        regex: /^.+@.+$/,
        stepNum: 0,
        message: "Please enter a valid email"
      },
      password: {
        regex: /^.{8,}$/,
        stepNum: 0,
        message: "Please enter a valid password (at least 8 characters)"
      },
      confirmPassword: {
        regex: new RegExp(
          `^${escapeStringRegexp(this.state.authAnswers.password)}$`
        ),
        stepNum: 0,
        forSignUp: true,
        message: "Please make sure your passwords match"
      },
      firstName: {
        regex: /^[^0-9]+$/,
        stepNum: 0,
        forSignUp: true,
        message: "Please enter a valid first name"
      },
      lastName: {
        regex: /^[^0-9]+$/,
        stepNum: 0,
        forSignUp: true,
        message: "Please enter a valid last name"
      },
      phoneNumber: {
        regex: /^.+$/,
        stepNum: 1,
        message: "Please enter a valid phone number"
      },
      birthday: {
        regex: /^[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]$/,
        stepNum: 1,
        message: "Please enter a valid date"
      },
      program: {
        regex: /^.+$/,
        stepNum: 1,
        message: "Please enter a valid program"
      },
      graduationYear: {
        regex: /^[0-9]{4}$/,
        stepNum: 1,
        message: "Please enter a valid year in the form YYYY"
      },
      whyApply: {
        regex: /^\s*[-\w]+(?:\W+[-\w]+){0,300}\W*\s*$/,
        stepNum: 2,
        message: "Please enter a valid answer (max 300 words)"
      },
      passionProject: {
        regex: /^\s*[-\w]+(?:\W+[-\w]+){0,300}\W*\s*$/,
        stepNum: 2,
        message: "Please enter a valid answer (max 300 words)"
      },
      dreamProject: {
        regex: /^\s*[-\w]+(?:\W+[-\w]+){0,100}\W*\s*$/,
        stepNum: 2,
        message: "Please enter a valid answer (max 100 words)"
      },
      agreeMlhTosAndPrivacyPolicy: {
        regex: /^true$/,
        stepNum: 2,
        message:
          "Please accept MLH's terms of service and privacy policy to apply"
      },
      agreeMlhCodeConduct: {
        regex: /^true$/,
        stepNum: 2,
        message: "Please accept MLH's code of conduct to apply"
      },
      numAttendedHackathons: {
        regex: /^[0-9]+$/,
        stepNum: 2,
        message: "Please enter a valid number"
      },
      fromLocation: {
        regex: /^.+$/,
        stepNum: 2,
        message: "Please enter a valid location"
      },
      gender: {
        regex: /^[^0-9]+$/,
        stepNum: 1,
        message: "Please enter your gender"
      },
      race: {
        regex: /^[^0-9]+$/,
        stepNum: 1,
        message: "Please enter your race"
      },
      school: {
        regex: /^.+$/,
        stepNum: 1,
        message: "Please enter your school"
      },
      degree: {
        regex: /^.+$/,
        stepNum: 1,
        message: "Please enter your degree"
      },
      graduationMonth: {
        regex: /^.+$/,
        stepNum: 1,
        message: "Please enter a valid month"
      }
    };
  }

  submit() {
    // const answers = this.state.applicationAnswers;
    // send request
    // this.nextStep();
  }

  validApplicationField(field) {
    return Object.prototype.hasOwnProperty.call(
      this.state.applicationAnswers,
      field
    );
  }

  validAuthField(field) {
    return Object.prototype.hasOwnProperty.call(this.state.authAnswers, field);
  }

  setAuthAnswer(field, answer) {
    if (!this.validAuthField(field)) {
      throw new Error(`No such field "${field}" in sign up form`);
    }
    const { authAnswers } = this.state;
    authAnswers[field] = answer;
    this.validateAnswer(field, answer, true);
    this.setState({ authAnswers });
  }

  setApplicationAnswer(field, answer) {
    if (!this.validApplicationField(field)) {
      throw new Error(`No such field "${field}" in application form`);
    }
    const { applicationAnswers } = this.state;
    applicationAnswers[field] = answer;
    this.validateAnswer(field, answer);
    this.setState({ applicationAnswers });
  }

  setError(field, err, auth = false) {
    if (auth) {
      if (!this.validAuthField(field)) {
        throw new Error(`No such field "${field} in authentication form`);
      }
    } else if (!this.validApplicationField(field)) {
      throw new Error(`No such field "${field} in application form`);
    }
    const { errors } = this.state;
    errors[field] = err;

    const hasErrors = Object.keys(errors).some((key) => errors[key]);

    this.setState({ errors, hasErrors });
  }

  nextStep() {
    if (this.validateAllAnswers()) {
      this.props.nextStep();
    }
  }

  previousStep() {
    this.props.previousStep();
  }

  validateAllAnswers() {
    let valid = true;

    Object.keys(this.state.applicationAnswers).forEach((field) => {
      valid =
        this.validateAnswer(field, this.state.applicationAnswers[field]) &&
        valid;
    });

    Object.keys(this.state.authAnswers).forEach((field) => {
      valid =
        this.validateAnswer(field, this.state.authAnswers[field], true) &&
        valid;
    });

    return valid;
  }

  validateAnswer(field, inAnswer, auth = false) {
    const validators = this.validators();
    const answer = String(inAnswer);
    if (auth) {
      if (!this.validAuthField(field)) {
        throw new Error(`No such field "${field}" in authentication form`);
      }
    } else if (!this.validApplicationField(field)) {
      throw new Error(`No such field "${field}" in application form`);
    }

    if (this.state.returningHacker && validators[field].forSignUp) {
      this.setError(field, "", auth);
      return true;
    }

    if (
      this.props.stepNum !== validators[field].stepNum ||
      validators[field].regex.test(answer)
    ) {
      this.setError(field, "", auth);
      return true;
    }

    this.setError(field, validators[field].message, auth);
    return false;
  }

  changeSelected(i) {
    const { errors } = this.state;
    Object.keys(errors).forEach((key) => {
      errors[key] = "";
    });
    this.setState({ returningHacker: i === 1, hasErrors: false, errors });
  }

  resetConfirmPassword() {
    const { authAnswers, errors } = this.state;
    authAnswers.confirmPassword = "";
    errors.confirmPassword = "";
    this.setState({ authAnswers, errors });
  }

  getQuestions(num) {
    const formStyle = `
      padding-top: 64px;
      max-width: 800px;
      margin: 0 auto;
    `;

    const subsectionStyle = `
      display: flex;
      margin: 36px 0;
      :not(:last-child) {
        border-bottom: 2px solid ${steel};
      }
      padding-bottom: 64px;
    `;

    const subsectionTitleStyles = `
      width: 30%;
      color: #575757;
      padding-top: 24px;
      font-weight: 600;
      text-transform: uppercase;
    `;

    const subsectionContentStyles = `
      width: 70%;
    `;

    const inputRowStyle = `
      width: 100%;
      display: flex;
      flex-direction: row;
      margin-top:12px;
      align-items: stretch;
      label, input {
          display: block;
      }
      >div{
          flex-grow: 1;
          min-width: 50%;
          padding: 4px;
      }
    `;

    const authHeadingStyle = `
      font-size: 32px;
      margin-top: 45px !important;
      margin-bottom: -20px !important;
    `;

    const allStyles = {
      authHeadingStyle,
      subsectionStyle,
      inputRowStyle,
      formStyle,
      subsectionContentStyles,
      subsectionTitleStyles
    };

    switch (num) {
      default: {
        return (
          <Step1
            alert={this.state.alert}
            alertShown={this.state.alertShown}
            returningHacker={this.state.returningHacker}
            changeSelected={this.changeSelected}
            authAnswers={this.state.authAnswers}
            setAuthAnswer={this.setAuthAnswer}
            errors={this.state.errors}
            hasErrors={this.state.hasErrors}
            nextStep={this.nextStep}
            resetConfirmPassword={this.resetConfirmPassword}
            login={this.login}
            createAccount={this.createAccount}
            {...allStyles}
          />
        );
      }
      case 1: {
        return (
          <Step2
            setApplicationAnswer={this.setApplicationAnswer}
            applicationAnswers={this.state.applicationAnswers}
            errors={this.state.errors}
            hasErrors={this.state.hasErrors}
            nextStep={this.nextStep}
            {...allStyles}
          />
        );
      }
      case 2: {
        return (
          <Step3
            setApplicationAnswer={this.setApplicationAnswer}
            applicationAnswers={this.state.applicationAnswers}
            errors={this.state.errors}
            hasErrors={this.state.hasErrors}
            previousStep={this.previousStep}
            submit={this.submit}
            {...allStyles}
          />
        );
      }
      case 3: {
        return <Step4 {...allStyles} />;
      }
    }
  }

  render() {
    return (
      <ContentWrapper>{this.getQuestions(this.props.stepNum)}</ContentWrapper>
    );
  }
}

export default compose(
  graphql(LOGIN_MUTATION, {
    name: "login"
  })
)(ApplicationForm);
