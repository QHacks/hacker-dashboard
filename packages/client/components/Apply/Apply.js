import { graphql, compose } from "react-apollo";
import { Redirect } from "react-router-dom";
import React, { Component } from "react";
import gql from "graphql-tag";

import MenuBar from "../MenuBar/index";
import ApplicationForm from "./ApplicationForm";
import ApplicationHeader from "./ApplicationHeader";
import ApplicationNavigation from "./ApplicationSteps/index";

const GET_AUTHENTICATION_STATUS = gql`
  query {
    authInfo @client {
      isAuthenticated
    }
  }
`;

const GET_HACKER_INFORMATION = gql`
  query {
    user {
      ... on Hacker {
        hasApplied(eventSlug: "qhacks-2019")
      }
    }
  }
`;

class Apply extends Component {
  constructor(props) {
    super(props);

    this.state = {
      stepNum: 0
    };
  }

  componentDidMount() {
    const { authInfo, hackerInfo } = this.props;

    if (authInfo && hackerInfo) {
      const { isAuthenticated } = authInfo;

      if (isAuthenticated) {
        this.setState({
          stepNum: 1
        });
      }
    }
  }

  previousStep() {
    this.setState((prevState) => ({ stepNum: prevState.stepNum - 1 }));
  }

  nextStep() {
    this.setState((prevState) => ({ stepNum: prevState.stepNum + 1 }));
  }

  render() {
    const {
      authInfo: { isAuthenticated },
      hackerInfo: { hasApplied },
      loadingHacker,
      loadingAuth
    } = this.props;

    if (loadingHacker || loadingAuth) {
      return <div>Loading...</div>;
    }

    if (isAuthenticated && hasApplied) {
      return (
        <Redirect
          to={{
            pathname: "/profile",
            state: {
              from: this.props.location,
              alert: {
                type: "info",
                message: "You've already applied to QHacks 2019!",
                status: "Yay!"
              }
            }
          }}
        />
      );
    }

    return (
      <div
        css={`
          h1,
          h2,
          h3 {
            color: black;
          }
        `}
      >
        <MenuBar showLogin hideItems={this.state.stepNum > 0} />
        <ApplicationHeader />
        <ApplicationNavigation stepNum={this.state.stepNum} />
        <ApplicationForm
          previousStep={() => this.previousStep()}
          nextStep={() => this.nextStep()}
          stepNum={this.state.stepNum}
        />
      </div>
    );
  }
}

const clientQuery = graphql(GET_AUTHENTICATION_STATUS, {
  props: ({ data }) => ({
    authInfo: data.authInfo,
    loadingAuth: data.loading
  })
});

const hackerQuery = graphql(GET_HACKER_INFORMATION, {
  options: {
    fetchPolicy: "cache-and-network"
  },
  props: ({ data }) => ({
    hackerInfo: data.user,
    loadingHacker: data.loading
  })
});

export default compose(
  clientQuery,
  hackerQuery
)(Apply);
