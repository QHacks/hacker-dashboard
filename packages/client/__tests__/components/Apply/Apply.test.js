import React from "react";
import Apply from "../../../components/Apply/Apply";
import { shallow } from "enzyme";

describe("Apply", () => {
  it("renders correctly", () => {
    expect(shallow(<Apply />)).toMatchSnapshot();
  });
});
