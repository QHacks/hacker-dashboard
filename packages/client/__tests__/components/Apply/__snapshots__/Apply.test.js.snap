// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apply renders correctly 1`] = `
<Query
  displayName="Apollo(Apollo(Apply))"
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": undefined,
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [
                  Object {
                    "arguments": Array [],
                    "kind": "Directive",
                    "name": Object {
                      "kind": "Name",
                      "value": "client",
                    },
                  },
                ],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "authInfo",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isAuthenticated",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 66,
        "start": 0,
      },
    }
  }
  skip={false}
  warnUnhandledError={true}
>
  <Component />
</Query>
`;
